x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -771,14 +772,30 @@ static void writeDissect(PlanStreamOutput out, Dissect dissect) throws IOExcepti
     }
 
     static EsRelation readEsRelation(PlanStreamInput in) throws IOException {
-        return new EsRelation(in.readSource(), readEsIndex(in), readAttributes(in), in.readBoolean());
+        Source source = in.readSource();
+        EsIndex esIndex = readEsIndex(in);
+        List<Attribute> attributes = readAttributes(in);
+        EsSourceOptions esSourceOptions = null;
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_ES_SOURCE_OPTIONS) && in.readBoolean()) {
+            esSourceOptions = new EsSourceOptions(in);
+        }
+        boolean frozen = in.readBoolean();
+        return new EsRelation(source, esIndex, attributes, esSourceOptions, frozen);
```

Comment:
True, but then we'd have to serialize an "empty" EsSourceOptions every time, while this functionality would probably not be used all that often. We do that already in some places (like serialising the source, which we send across just in case we generate Warnings), and while an empty EsSourceOptions is quite light (just three `false`s currently), I thought I'd avoid it.
Then, EsRelation would need to create this object in the c'tors that don't receive it.

Reply:
See above - I'll resolve this to avoid discussing in 2 places :)